app/scripts/modules/ecs/src/serverGroup/configure/wizard/capacityProvider/CapacityProvider.tsx

```diff
@@ -19,10 +27,42 @@ class EcsCapacityProvider extends React.Component<IEcsCapacityProviderProps, IEc
     const cmd = this.props.command;
 
     this.state = {
-      capacityProviderStrategy: cmd.capacityProviderStrategy,
+      availableCapacityProviders: this.setAvailableCapacityProviders(cmd),
+      defaultCapacityProviderStrategy: this.setDefaultCapacityProviderStrategy(cmd),
+      ecsClusterName: cmd.ecsClusterName,
+      useDefaultCapacityProviders: cmd.useDefaultCapacityProviders || cmd.capacityProviderStrategy && cmd.capacityProviderStrategy.length == 0,
+      capacityProviderStrategy: cmd.capacityProviderStrategy.length > 0 ? cmd.capacityProviderStrategy : [],
+      capacityProviderLoadedFlag: false
```

Comment:
What's the purpose behind this flag? Can we use one of the other state properties to determine if it's loaded?

Reply:
Good question, I had thought of choosing other state property as a candidate for determining if it's loaded or not, but other state properties are sometimes populated but sometimes empty. That's the reason we can not use them to as a deciding factor over here. I hope this answers your concern.

Reply:
What does your stage json look like ? (when you click edit stage as json after you add capacity providers).

Reply:
{
  "clusters": [
    {
      "account": "ecs-my-aws-devel-acct",
      "application": "aws",
      "associatePublicIpAddress": true,
      "availabilityZones": {
        "us-west-2": [
          "us-west-2a",
          "us-west-2b",
          "us-west-2c",
          "us-west-2d"
        ]
      },
      "capacity": {
        "desired": 2,
        "max": 63,
        "min": 1
      },
      "capacityProviderStrategy": [
        {
          "base": 1,
          "capacityProvider": "FARGATE",
          "weight": 1
        },
        {
          "base": 1,
          "capacityProvider": "FARGATE_SPOT",
          "weight": 2
        }
      ],
      "cloudProvider": "ecs",
      "computeUnits": 256,
      "containerPort": 80,
      "copySourceScalingPoliciesAndActions": false,
      "delayBeforeDisableSec": 0,
      "delayBeforeScaleDownSec": 0,
      "dockerImageCredentialsSecret": "None (No registry credentials)",
      "dockerLabels": {},
      "ecsClusterName": "example-app-test",
      "environmentVariables": {
        "PORT": "80"
      },
      "freeFormDetails": "ecsdemo",
      "healthCheckGracePeriodSeconds": 5,
      "healthCheckType": "EC2",
      "iamRole": "None (No IAM role)",
      "imageDescription": {
        "account": "my-us-west-2-devel-registry",
        "fromTrigger": true,
        "imageId": "123123123dkr.ecr.us-west-2.amazonaws.com/spinnaker-deployment-images (Tag resolved at runtime)",
        "registry": "123123123.dkr.ecr.us-west-2.amazonaws.com",
        "repository": "spinnaker-deployment-images",
        "tag": ""
      },
      "launchType": "",
      "loadBalancedContainer": "",
      "loadBalancers": [],
      "logDriver": "awslogs",
      "logOptions": {
        "awslogs-group": "spinnaker-ecs-demo",
        "awslogs-region": "us-west-2",
        "awslogs-stream-prefix": "spinnaker"
      },
      "maxRemainingAsgs": 2,
      "moniker": {
        "app": "aws",
        "detail": "ecsdemo",
        "stack": "prod"
      },
      "networkMode": "awsvpc",
      "placementConstraints": [],
      "placementStrategyName": "",
      "placementStrategySequence": [],
      "preferSourceCapacity": true,
      "provider": "ecs",
      "reservedMemory": 512,
      "rollback": {
        "onFailure": true
      },
      "scaleDown": true,
      "securityGroupNames": [
        "spinnaker-ecs-demo-private-access"
      ],
      "securityGroups": [],
      "serviceDiscoveryAssociations": [],
      "stack": "prod",
      "strategy": "redblack",
      "subnetType": "",
      "subnetTypes": [
        "public-subnet"
      ],
      "tags": {},
      "targetGroup": "",
      "targetGroupMappings": [
        {
          "containerName": "",
          "containerPort": 80,
          "targetGroup": "spinnaker-ecs-demo-tg"
        }
      ],
      "taskDefinitionArtifact": {},
      "useDefaultCapacityProviders": false,
      "useSourceCapacity": false,
      "useTaskDefinitionArtifact": false
    }
  ],
  "name": "Deploy",
  "type": "deploy"
}